### PR TITLE
(BKR-220) test tagging implemented

### PR DESCRIPTION
**Pull request by kevpl**

### CODE REVIEW

**Comment by kevpl** [comment]
it appears there are more than unrelated failures here. Looking, will update the ticket when I have more.

**Comment by kevpl** [comment]
looks like Jenkins was just having git issues.  retrying

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by kevpl** [comment]
@puppetlabs-jenkins retest this please

**Comment by anodelman** [comment]
You are going to need to rebase this to get the git acceptance test pin fix.

**Comment by kevpl** [comment]
rebased

**Comment by kevpl** [comment]
doesn't pass spec after rebase.  looking at this now...

**Comment by kevpl** [comment]
had an issue w/using preserved host files. It was treating them as unsanitary inputs, trying to normalize them as it would user input.  Now, if it's using a preserved host file, it'll see that they're already in the sanitized type and skip that step, but still do the rest of normalization and validation correctly.

**Comment by anodelman** [comment]
Please write a new wiki doc to cover this functionality.

File: lib/beaker/options/parser.rb [hunk 1]
```diff
@@ -345,6 +347,29 @@ def normalize_args
...
+        @options[:tag_includes] = @options[:tag_includes].split(',') if @options[:tag_includes].respond_to?(:split)
+        @options[:tag_excludes] = @options[:tag_excludes].split(',') if @options[:tag_excludes].respond_to?(:split)
+        @options[:tag_includes].map!(&:downcase)
+        @options[:tag_excludes].map!(&:downcase)
+
+        #check that a tag is not both included & excluded sets
+        @options[:tag_includes].each do |included_tag|
+          @options[:tag_excludes].each do |excluded_tag|
+            parser_error "tag '#{included_tag}' cannot be in both the included and excluded tag sets" \
+              if included_tag == excluded_tag
```
Comment (waynr) [inline lib/beaker/options/parser.rb]: Another approach would be to check for set intersection here:
http://ruby-doc.org/stdlib-2.2.2/libdoc/set/rdoc/Set.html#method-i-intersect-3F

**Comment by waynr** [comment]
:see_no_evil:, :+1: after updating docs.

**Comment by kevpl** [comment]
added [doc](https://github.com/puppetlabs/beaker/wiki/Beaker-Test-Tagging)

**Pull request merged**